src/GoTrueClient.ts

```diff
@@ -330,6 +335,39 @@ export default class GoTrueClient {
     return this.currentSession
   }
 
+  async getSession(): Promise<
+    | {
+        session: Session
+        error: null
+      }
+    | {
+        session: null
+        error: ApiError
+      }
+    | {
+        session: null
+        error: null
+      }
+  > {
+    if (!this.currentSession) {
+      return { session: null, error: null }
+    }
+
+    const hasExpired = this.currentSession.expires_at
+      ? this.currentSession.expires_at <= Date.now() / 1000
```

Comment:
have some threshold here? we already have a constate for this i think

Reply:
Is it needed here? If this handler works for only refreshing the token if it's already somehow expired, and the regular refresh timer handles refreshing tokens early, users rarely have to wait for the refresh request before their normal request.

Not opposed to it, just wondering what the benefits are vs the drawbacks?